### DISCUSSION TITLE
Please use `model_validate_json()` for JSON in request body

### DISCUSSION THREAD

**Discussion by methane** [fastapi/fastapi#9868] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Body
from pydantic import BaseModel, ConfigDict
from enum import IntEnum


app = FastAPI()


class Color(IntEnum):
    blue = 1
    red = 2


class Item(BaseModel):
    name: str
    color: Color
    model_config = ConfigDict(strict=True)


@app.post("/item")
async def create_item(item: Item = Body()):
    print(repr(item))
    return {}

if __name__ == '__main__':
    print(repr(Item.model_validate_json('{"name":"cup", "color": 1}')))
```

### Description

Pydantic v2 has model_validate_json, that parses JSON and validate it with JSON mode.
FastAPI 0.100.0 doesn't use it to parse JSON for now. So it is difficult to use strict mode for models.

```
uvicorn app:app --reload
...
> curl -X 'POST' \
        'http://127.0.0.1:8000/item' \
        -H 'accept: application/json' \
        -H 'Content-Type: application/json' \
        -d '{
    "name": "cup",
    "color": 1
  }' | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   232  100   199  100    33  27539   4566 --:--:-- --:--:-- --:--:--  226k
{
  "detail": [
    {
      "type": "is_instance_of",
      "loc": [
        "body",
        "color"
      ],
      "msg": "Input should be an instance of Color",
      "input": 1,
      "ctx": {
        "class": "Color"
      },
      "url": "https://errors.pydantic.dev/2.1.2/v/is_instance_of"
    }
  ]
}
```

### Operating System

Linux, macOS

### FastAPI Version

0.100.0

### Python Version

3.11.3

**Comment by ghost** [comment]
At the moment, the following implementation using [field_validator](https://docs.pydantic.dev/latest/usage/validators/) with 'before' might be a workaround:

```
class Item(BaseModel):
    name: str
    color: Color
    model_config = ConfigDict(strict=True)

    @field_validator('color', mode='before')
    def value_to_enum(cls, v):
        if isinstance(v, int):
            return Color(v)
        return v
```

**Comment by PaulVanSchayck** [comment]
This seems still to be the case with FastAPI 0.103.1. The workaround resolves it, but is a bit ugly. Should we create an issue for this? I think it's a bug.

**Comment by TomerGodinger** [comment]
It is still present in FastAPI 0.111.1.

@tiangolo Can we please make this into an issue? It's problematic that you can't use enums with strict validation, and the workaround here needs to be added to every class that includes an enum (creating a general solution in a base class that covers all cases is... problematic, at best).

**Comment by YuriiMotov** [comment, marked as answer]
Not sure it's possible to use `model_validate_json` here: you can add more than one Body model to your endpoint and this way will still need to parse JSON to extract data to pass to model validators..

Easier way to solve this is by adding `Field(strict=False)` annotation to such fields:

```python
class Item(BaseModel):
    name: str
    color: Annotated[Color, Field(strict=False)]
    model_config = ConfigDict(strict=True)
```
or
```python
ColorNonStrict: TypeAlias = Annotated[Color, Field(strict=False)]

class ItemV2(BaseModel):
    name: str
    color: ColorNonStrict
    model_config = ConfigDict(strict=True)

```

Tests:
```python
from typing import Annotated, TypeAlias
from fastapi import FastAPI, Body
from fastapi.testclient import TestClient
from pydantic import BaseModel, ConfigDict, Field
from enum import IntEnum

import pytest


app = FastAPI()

# V1 

class Color(IntEnum):
    blue = 1
    red = 2


class Item(BaseModel):
    name: str
    color: Annotated[Color, Field(strict=False)]
    model_config = ConfigDict(strict=True)

@app.post("/item")
async def create_item(item: Item = Body()):
    print(repr(item))
    return {}


# V2

ColorNonStrict: TypeAlias = Annotated[Color, Field(strict=False)]

class ItemV2(BaseModel):
    name: str
    color: ColorNonStrict
    model_config = ConfigDict(strict=True)

@app.post("/item-v2")
async def create_item_v2(item: ItemV2 = Body()):
    print(repr(item))
    return {}


# ============================================================================
# Test

@pytest.mark.parametrize("path", ["/item", "/item-v2"])
def test(path: str):
    client = TestClient(app)
    resp = client.post(path, json={"name": "string", "color": 1})
    assert resp.status_code == 200

```
